Make `ConverterManager.getInstance()` init thread-safe.

cpovirk opened the pull request:
We encountered a TSAN error because 2 threads were racing to call
`ConverterManager.getInstance()`. My read is that the current code could
actually be unsafe, since one thread might see `INSTANCE` as non-null
before the other thread has initialized its fields.

Fortunately, this looks like a good case for [the
initialization-on-demand holder
idiom](https://www.cs.umd.edu/~pugh/java/memoryModel/jsr-133-faq.html#:~:text=initialization%20on%20demand%20holder%20idiom).

(There would be additional thread-safety concerns if anyone were to try
to _mutate_ the `ConverterManager` instance concurrently with usage. But
I'm not sure there's a solution to that short of spraying `synchronized`
around. That seems like probably overkill, given the possible
performance impact for what I'd hope is an uncommon use case.)

jodastephen commented:
Looks like a good change and happy to merge, but unfortunately `TestConverterManager` needs fixing.

cpovirk commented:
The tests now pass.

I updated the assertion to look for the field in the new class. I now check that that _class_ is `private`, but I figured that whether the _field_ is `private` is mostly irrelevant in that case. Rather than throw away the field check entirely, I changed it to a check for `final`, since we're now able to use that.

But I'm happy to update it in some other way if you think that would be best.

jodastephen commented:
Thanks!
